specification/compute/Compute.Management/VirtualMachineDiagnosticRunCommand.tsp

```diff
@@ -0,0 +1,213 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "./models.tsp";
+import "./VirtualMachine.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+using Versioning;
+
+namespace Microsoft.Compute;
+/**
+ * Describes a Virtual Machine diagnostic run command.
+ */
+@added(Versions.v2025_04_01)
+@parentResource(VirtualMachine)
+model VirtualMachineDiagnosticRunCommand
+  is Azure.ResourceManager.TrackedResource<VirtualMachineRunCommandProperties> {
+  ...ResourceNameParameter<
+    Resource = VirtualMachineDiagnosticRunCommand,
+    KeyName = "runCommandName",
+    SegmentName = "diagnosticRunCommands",
+    NamePattern = ""
+  >;
+}
+
+@armResourceOperations
+interface VirtualMachineDiagnosticRunCommandOps
+  extends Azure.ResourceManager.Legacy.ComputeLegacyOperations<
+      VirtualMachinePath,
+      KeysOf<{
+        ...ResourceNameParameter<
+          VirtualMachineDiagnosticRunCommand,
+          SegmentName = "diagnosticRunCommands",
+          KeyName = "runCommandName",
+          NamePattern = ""
+        >,
+      }>
+    > {}
+
+@armResourceOperations
+interface VirtualMachineDiagnosticRunCommands {
+  /**
+   * The operation to get the diagnostic run command.
+   */
+  @added(Versions.v2025_04_01)
+  getByVirtualMachine is VirtualMachineDiagnosticRunCommandOps.Read<
+    VirtualMachineDiagnosticRunCommand,
+    Parameters = {
+      /**
+       * The expand expression to apply on the operation.
+       */
+      @query("$expand")
+      $expand?: string;
+    }
+  >;
+
+  /**
+   * The operation to create or update the diagnostic run command.
+   */
+  @added(Versions.v2025_04_01)
+  createOrUpdate is VirtualMachineDiagnosticRunCommandOps.CreateOrUpdateAsync<
+    VirtualMachineDiagnosticRunCommand,
+    LroHeaders = ArmLroLocationHeader & Azure.Core.Foundations.RetryAfterHeader
+  >;
+
+  /**
+   * The operation to update the diagnostic run command.
+   */
+  @added(Versions.v2025_04_01)
+  @patch(#{ implicitOptionality: false })
+  update is VirtualMachineDiagnosticRunCommandOps.CustomPatchAsync<
+    VirtualMachineDiagnosticRunCommand,
+    VirtualMachineRunCommandUpdate,
+    Response = ArmResponse<VirtualMachineDiagnosticRunCommand> &
```

Comment:
>ArmResponse [](http://example.com/codeflow?start=15&length=11)

200 is not the correct response for async PATCH (should be 202). Ref: [https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#updating-using-patch](https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#updating-using-patch).